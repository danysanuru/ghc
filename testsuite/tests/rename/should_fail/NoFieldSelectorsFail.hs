{-# LANGUAGE NoFieldSelectors #-}

module NoFieldSelectorsFail (foo)
where

import Prelude


data Foo = Foo { foo :: Int, bar :: String }
data Bar = Bar { foo :: Int, bar' :: String }

foo1 :: Foo
foo1 = Foo 3 "bar"

bar0 = Bar { foo = 0, bar' = "bar'" }

foo3 :: Foo
foo3 = foo1 { foo = 4 } -- update

bar1 = bar0 { foo = 1 }

bar = undefined

foo4 = foo1 { bar = "" } -- currently rejected, see #18999
