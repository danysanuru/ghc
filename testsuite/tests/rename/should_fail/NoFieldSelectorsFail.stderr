
NoFieldSelectorsFail.hs:15:14: error:
    Ambiguous occurrence ‘foo’
    It could refer to
       either the field ‘foo’, defined at NoFieldSelectorsFail.hs:10:18
           or the field ‘foo’, defined at NoFieldSelectorsFail.hs:9:18

NoFieldSelectorsFail.hs:18:15: error:
    Ambiguous occurrence ‘foo’
    It could refer to
       either the field ‘foo’, defined at NoFieldSelectorsFail.hs:10:18
           or the field ‘foo’, defined at NoFieldSelectorsFail.hs:9:18

NoFieldSelectorsFail.hs:20:15: error:
    Ambiguous occurrence ‘foo’
    It could refer to
       either the field ‘foo’, defined at NoFieldSelectorsFail.hs:10:18
           or the field ‘foo’, defined at NoFieldSelectorsFail.hs:9:18

NoFieldSelectorsFail.hs:24:15: error:
    Ambiguous occurrence ‘bar’
    It could refer to
       either the field ‘bar’, defined at NoFieldSelectorsFail.hs:9:30
           or ‘NoFieldSelectorsFail.bar’,
              defined at NoFieldSelectorsFail.hs:22:1
